AtReconstruction/AtFitter/OpenKF/kalman_filter/TrackFitterUKFTest.cxx

```diff
@@ -0,0 +1,268 @@
+
+   ASSERT_NEAR(m_ukf.matP()(0, 0), 0.01433114F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(0, 1), -0.01026142F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(0, 2), 0.00651178F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(0, 3), -0.00465059F, FLOAT_EPSILON);
+
+   ASSERT_NEAR(m_ukf.matP()(1, 0), -0.01026142F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(1, 1), 0.0295458F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(1, 2), -0.0046378F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(1, 3), 0.01344241F, FLOAT_EPSILON);
+
+   ASSERT_NEAR(m_ukf.matP()(2, 0), 0.00651178F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(2, 1), -0.0046378F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(2, 2), 0.13023154F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(2, 3), -0.00210188F, FLOAT_EPSILON);
+
+   ASSERT_NEAR(m_ukf.matP()(3, 0), -0.00465059F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(3, 1), 0.01344241F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(3, 2), -0.00210188F, FLOAT_EPSILON);
+   ASSERT_NEAR(m_ukf.matP()(3, 3), 0.1333886F, FLOAT_EPSILON);
+}
+
+class TrackFitterUKFPhysicsTest : public testing::Test {
+public:
+   virtual void SetUp() override {}
+   virtual void TearDown() override {}
+
+   static constexpr float FLOAT_EPSILON{0.001F};
+
+   static constexpr size_t DIM_X{6};
+   static constexpr size_t DIM_V{2};
+   static constexpr size_t DIM_Z{3};
+   static constexpr size_t DIM_N{3};
+
+   kf::TrackFitterUKF<DIM_X, DIM_Z, DIM_V, DIM_N> m_ukf;
+
+   /// @brief to propagate the state vector using the process model
+   /// @param x state vector
+   /// @param v process noise vector
+   /// @return propagated (unaugmented) state vector
+   static kf::Vector<DIM_X> funcF(const kf::Vector<DIM_X> &x, const kf::Vector<DIM_V> &v)
+   {
+      // TODO: This needs to be filled with an RK4 solver for the physics model
+      kf::Vector<DIM_X> y{x};
+
+      // For now, we just return the state vector as is
+      return y;
+   }
+
+   /// @brief to apply the measurement model to the state vector
+   /// @param x the state vector of the system
+   /// @return the measurement vector
+   static kf::Vector<DIM_Z> funcH(const kf::Vector<DIM_X> &x)
+   {
+      kf::Vector<DIM_Z> y;
+      y[0] = x[0];
+      y[1] = x[1];
+      y[2] = x[2];
+
+      return y;
+   }
+};
+
+TEST_F(TrackFitterUKFPhysicsTest, PhysicsPrediction)
+{
+   // TODO: This needs to be filled with a proper physics test.
+   kf::Vector<DIM_X> x; // Initial state vector
+
+   kf::Matrix<DIM_X, DIM_X> P; // Initial state vector covariance matrix
+
+   // Note: process noise is defined in the UKF class, so we don't need to set it here.
+
+   kf::Vector<DIM_Z> z; // Measurement vector to be used in the correction step
+   z << std::sqrt(5), std::atan2(1.f, 2.f);
+
+   kf::Matrix<DIM_N, DIM_N> R; // Covariance matrix for the measurement noise
+
+   m_ukf.vecX() = x;
+   m_ukf.matP() = P;
+
+   m_ukf.setCovarianceR(R);
+   m_ukf.predictUKF(funcF);
+}
```

Comment:
@Yassid Once we have the propagator, we will need to fill in the rest of this test

We may also want to add some specific tets cases for the `funcF` method